etna/transforms/decomposition/detrend.py

```diff
@@ -40,14 +60,11 @@ def fit(self, df: pd.DataFrame) -> "_OneSegmentLinearTrendBaseTransform":
             instance with trained regressor
         """
         df = df.dropna(subset=[self.in_column])
-        series_len = len(df)
-        x = df.index.to_series()
-        if isinstance(type(x.dtype), pd.Timestamp):
-            raise ValueError("Your timestamp column has wrong format. Need np.datetime64 or datetime.datetime")
-        x = x.apply(lambda ts: ts.timestamp())
-        x = x.to_numpy().reshape(series_len, 1)
+        x = self._get_x(df)
+        self._x_median = np.median(x)
```

Comment:
It changes default behaviour of detrending. 
Any reason to do such things? Why median, why not mean-std scaling that should be reasonable for reggresions?

Reply:
The reason is connected to the fact that during polynomial transformations we meet timestamp**2 and `TheilSenTrendTransform` can't work with it properly (it gives very strange results). I think it is because around 2020-th year we have values around 1e9 and after squaring they become 1e18.
If we substract median, then values become much lower and computational problem leaves.